modules/gapi/src/compiler/passes/pattern_matching.cpp

```diff
@@ -0,0 +1,511 @@
+                       std::vector<ade::NodeHandle>, // test graph matched nodes for the pattern OP
+                       ade::HandleHasher<ade::Node>> allMatchingsForFirstOpNodes;
+
+    //Fit the allMatchingsForFirstOpNodes
+
+    std::size_t possibleStartPointsCount = 1;
+
+    // For every starting OP node of pattern identify matching candidates in test graph.
+    // For every starting pattern node there may be multiple matching candidates.
+    auto testNodes = testGraph.nodes();
+    for (auto firstPatternOpNode : firstPatternOpNodes) {
+        auto firstMetadata = patternGraph.metadata(firstPatternOpNode);
+
+        std::vector<ade::NodeHandle> possibleMatchings;
+        std::copy_if(testNodes.begin(), testNodes.end(), std::back_inserter(possibleMatchings),
+            [&](const ade::NodeHandle& node) {
+            auto secondMetadata = testGraph.metadata(node);
+            bool stub = false;
+            /* TODO: FIXX */
+
+            if (secondMetadata.get<cv::gimpl::NodeType>().t == cv::gimpl::NodeType::OP) {
+                return opNodesComparator({ },
+                                         firstPatternOpNode, {  }, firstMetadata,
+                                         node, {  }, secondMetadata,
+                                         stub);
+            }
+            else {
+                return false;
+            }
+        });
+
+        possibleStartPointsCount *= possibleMatchings.size();
+        allMatchingsForFirstOpNodes[firstPatternOpNode] = std::move(possibleMatchings);
+    }
+
+    if (possibleStartPointsCount == 0) {
+        // Pattern graph is not matched
+        return SubgraphMatch { };
+    }
+    // Bad namings
+    // TODO FIX: Use using
+    SubgraphMatch::M subgraphStartOps;
+    SubgraphMatch::M subgraphEndOps;
+    // FIXME: consider moving to S
+    std::list<ade::NodeHandle> subgraphInternals;
+
+
+    // Structural matching first, semantic matching second.
+
+    //TODO: found means pattern is matched except input and output DATA nodes
+    bool found = false;
+    std::size_t i = 0;
+    while (!found && (i < possibleStartPointsCount)) {
+        subgraphStartOps.clear();
+        subgraphEndOps.clear();
+        subgraphInternals.clear();
+
+        // For visited matchings to work correctly it is required that output nodes from some node found in both graphs were in the same order!
+        // And that outNodes array contains these nodes right in the order that they are presented in graph (left-to-right nodes-exact).
+        // Else graphs were not assumed to be equal.
+        VisitedMatchings matchedVisitedNodes;
+
+        // This loop pushes the next combination from the cartesian product (identified by i, see comment below)
+        // to matchedVisitedNodes list.
+        // As a result, matchedVisitedNodes contains matching candidates  for the current vector of starting nodes.
+        std::size_t div = i;
```

Comment:
there's `div` function in C/C++, please rename the variable

Reply:
Done